### PR TITLE
Backports/jammy

### PR DESCRIPTION
**Pull request by ArrayBolt3**
Finished backport, all architectures are perfectly happy with the symbols, also got Lintian took care of.

### CODE REVIEW

File: debian/libfm-qt11.symbols [hunk 1]
```diff
@@ -368,6 +368,10 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm12FolderConfig10getIntegerEPKcPi@Base 0.16.0
  _ZN2Fm12FolderConfig10setBooleanEPKcb@Base 0.16.0
  _ZN2Fm12FolderConfig10setIntegerEPKci@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm12FolderConfig13getStringListEPKcPj@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:371]: I would be careful here. Compare the c++filt output between 371 and 372, same with 373 and 374. I suspect the symbols will fail once this change hits the archive. Remind me to use a CI train PPA to test.

File: debian/libfm-qt11.symbols [hunk 2]
```diff
@@ -377,10 +381,14 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm12FolderConfig8finalizeEv@Base 0.16.0
  _ZN2Fm12FolderConfig9getDoubleEPKcPd@Base 0.16.0
  _ZN2Fm12FolderConfig9getStringEPKc@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm12FolderConfig9getUint64EPKcPy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:384]: Ditto

File: debian/libfm-qt11.symbols [hunk 3]
```diff
@@ -377,10 +381,14 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
...
+ (arch=!riscv64 !i386 !armhf)_ZN2Fm12FolderConfig9getUint64EPKcPm@Base 1.1.0
  _ZN2Fm12FolderConfig9removeKeyEPKc@Base 0.16.0
  _ZN2Fm12FolderConfig9saveCacheEv@Base 0.16.0
  _ZN2Fm12FolderConfig9setDoubleEPKcd@Base 0.16.0
  _ZN2Fm12FolderConfig9setStringEPKcS2_@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm12FolderConfig9setUint64EPKcy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:390]: Ditto

File: debian/libfm-qt11.symbols [hunk 4]
```diff
@@ -413,6 +421,8 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm12ThumbnailJob16staticMetaObjectE@Base 0.16.0
  _ZN2Fm12ThumbnailJob17generateThumbnailERKSt10shared_ptrIKNS_8FileInfoEERKNS_8FilePathEPKcRK7QString@Base 0.16.0
  _ZN2Fm12ThumbnailJob17setLocalFilesOnlyEb@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm12ThumbnailJob19readImageFromStreamEP13_GInputStreamj@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:424]: Ditto

File: debian/libfm-qt11.symbols [hunk 5]
```diff
@@ -541,6 +551,8 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm14MountOperation20onEjectMountFinishedEP7_GMountP13_GAsyncResultP8QPointerIS0_E@Base 0.16.0
  _ZN2Fm14MountOperation21onEjectVolumeFinishedEP8_GVolumeP13_GAsyncResultP8QPointerIS0_E@Base 0.16.0
  _ZN2Fm14MountOperation21onMountVolumeFinishedEP8_GVolumeP13_GAsyncResultP8QPointerIS0_E@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm14MountOperation21onShowUnmountProgressEP16_GMountOperationPcxxPS0_@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:554]: Ditto

File: debian/libfm-qt11.symbols [hunk 6]
```diff
@@ -553,6 +565,8 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm14MountOperationD1Ev@Base 0.16.0
  _ZN2Fm14MountOperationD2Ev@Base 0.16.0
  _ZN2Fm14changeFileNameERKNS_8FilePathERK7QStringP7QWidgetb@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm14formatFileSizeEyb@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:568]: Ditto

File: debian/libfm-qt11.symbols [hunk 7]
```diff
@@ -586,6 +600,8 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm15FileTransferJob15copyRegularFileERKNS_8FilePathERKNS_10GObjectPtrI10_GFileInfoEERS1_@Base 0.16.0
  _ZN2Fm15FileTransferJob15copySpecialFileERKNS_8FilePathERKNS_10GObjectPtrI10_GFileInfoEERS1_@Base 0.16.0
  _ZN2Fm15FileTransferJob16staticMetaObjectE@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm15FileTransferJob25gfileCopyProgressCallbackExxPS0_@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:603]: Ditto

File: debian/libfm-qt11.symbols [hunk 8]
```diff
@@ -679,15 +695,25 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm16FileOperationJob11qt_metacastEPKc@Base 0.16.0
  _ZN2Fm16FileOperationJob13preparedToRunEv@Base 0.16.0
  _ZN2Fm16FileOperationJob14setCurrentFileERKNS_8FilePathE@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm16FileOperationJob14setTotalAmountEyy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:698]: Ditto

File: debian/libfm-qt11.symbols [hunk 9]
```diff
@@ -679,15 +695,25 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
...
+ (arch=!riscv64 !i386 !armhf)_ZN2Fm16FileOperationJob14setTotalAmountEmm@Base 1.1.0
  _ZN2Fm16FileOperationJob16staticMetaObjectE@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm16FileOperationJob17addFinishedAmountEyy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:701]: Ditto

File: debian/libfm-qt11.symbols [hunk 10]
```diff
@@ -679,15 +695,25 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
...
+ (arch=!riscv64 !i386 !armhf)_ZN2Fm16FileOperationJob17addFinishedAmountEmm@Base 1.1.0
+ (arch=i386 armhf)_ZN2Fm16FileOperationJob17setFinishedAmountEyy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:703]: Ditto

File: debian/libfm-qt11.symbols [hunk 11]
```diff
@@ -679,15 +695,25 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
...
+ (arch=!riscv64 !i386 !armhf)_ZN2Fm16FileOperationJob17setFinishedAmountEmm@Base 1.1.0
+ (arch=i386 armhf)_ZN2Fm16FileOperationJob22setCurrentFileProgressEyy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:705]: Ditto

File: debian/libfm-qt11.symbols [hunk 12]
```diff
@@ -824,6 +850,10 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm19FileOperationDialog14setSourceFilesERKSt6vectorINS_8FilePathESaIS2_EE@Base 0.16.0
  _ZN2Fm19FileOperationDialog16setRemainingTimeEj@Base 0.16.0
  _ZN2Fm19FileOperationDialog16staticMetaObjectE@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm19FileOperationDialog17setFilesProcessedEyy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:853]: Ditto

File: debian/libfm-qt11.symbols [hunk 13]
```diff
@@ -824,6 +850,10 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
...
+ (arch=!riscv64 !i386 !armhf)_ZN2Fm19FileOperationDialog17setFilesProcessedEmm@Base 1.1.0
+ (arch=i386 armhf)_ZN2Fm19FileOperationDialog18setDataTransferredEyy@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:855]: Ditto

File: debian/libfm-qt11.symbols [hunk 14]
```diff
@@ -1239,10 +1269,18 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZNK2Fm16FileDialogHelper9directoryEv@Base 0.16.0
  _ZNK2Fm16FileOperationJob10metaObjectEv@Base 0.16.0
  _ZNK2Fm16FileOperationJob11currentFileEv@Base 0.16.0
+ (arch=i386 armhf)_ZNK2Fm16FileOperationJob11totalAmountERyS1_@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:1272]: Ditto

File: debian/libfm-qt11.symbols [hunk 15]
```diff
@@ -1239,10 +1269,18 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
...
+ (arch=!riscv64 !i386 !armhf)_ZNK2Fm16FileOperationJob11totalAmountERmS1_@Base 1.1.0
+ (arch=i386 armhf)_ZNK2Fm16FileOperationJob14finishedAmountERyS1_@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:1274]: Ditto

File: debian/libfm-qt11.symbols [hunk 16]
```diff
@@ -1239,10 +1269,18 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
...
+ (arch=!riscv64 !i386 !armhf)_ZNK2Fm16FileOperationJob14finishedAmountERmS1_@Base 1.1.0
+ (arch=i386 armhf)_ZNK2Fm16FileOperationJob19currentFileProgressERNS_8FilePathERyS3_@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:1276]: Ditto

File: debian/libfm-qt11.symbols [hunk 17]
```diff
@@ -1276,6 +1314,8 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZNK2Fm6Folder10metaObjectEv@Base 0.16.0
  _ZNK2Fm6Folder13isIncrementalEv@Base 0.16.0
  _ZNK2Fm6Folder14hasFileMonitorEv@Base 0.16.0
+ (arch=i386 armhf)_ZNK2Fm6Folder17getFilesystemInfoEPyS1_@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:1317]: Ditto

File: debian/libfm-qt11.symbols [hunk 18]
```diff
@@ -1299,6 +1339,8 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZNK2Fm9Bookmarks13bookmarksFileEv@Base 0.16.0
  _ZNK2Fm9DeleteJob10metaObjectEv@Base 0.16.0
  _ZNK2Fm9Templates10metaObjectEv@Base 0.16.0
+ (optional=templinst|arch=i386 armhf)_ZNKSt10_HashtableINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4pairIKS5_St10shared_ptrIKN2Fm8FileInfoEEESaISD_ENSt8__detail10_Select1stESt8equal_toIS5_ESt4hashIS5_ENSF_18_Mod_range_hashingENSF_20_Default_ranged_hashENSF_20_Prime_rehash_policyENSF_17_Hashtable_traitsILb1ELb0ELb1EEEE19_M_find_before_nodeEjRS7_j@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:1342]: Ditto, nice job on finding `optional=templinst` :smile:

File: debian/libfm-qt11.symbols [hunk 19]
```diff
@@ -1626,6 +1668,8 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  (arch=riscv64)_ZTVSt23_Sp_counted_ptr_inplaceIN2Fm9TemplatesESaIS1_ELN9__gnu_cxx12_Lock_policyE1EE@Base 1.1.0
  (arch=!riscv64)_ZTVSt23_Sp_counted_ptr_inplaceIN2Fm9TemplatesESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE@Base 0.16.0
  _ZZNSt19_Sp_make_shared_tag5_S_tiEvE5__tag@Base 0.16.0
+ (optional=templinst|arch=i386 armhf)_ZZNSt8__detail18__to_chars_10_implIjEEvPcjT_E8__digits@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:1671]: Ditto

File: debian/copyright [hunk 20]
```diff
@@ -17,7 +17,7 @@ Copyright: 2015 Luís Pereira <luis.artur.pereira@gmail.com>
 License: BSD-3-Clause
 
 Files: debian/*
-Copyright: 2021, Lubuntu Developers <lubuntu-devel@lists.ubuntu.com>
+Copyright: 2021-2022, Lubuntu Developers <lubuntu-devel@lists.ubuntu.com>
```
Comment (tsimonq2) [inline debian/copyright:20]: Please confirm that this is the only copyright change needed.

File: debian/changelog [hunk 21]
```diff
@@ -1,3 +1,14 @@
+libfm-qt (1.1.0-0ubuntu7~ppa2) jammy; urgency=medium
+
+  * Backport to Jammy.
+  * Updated Lubuntu Developers copyright.
+  * Updated symbols file
+  * Removed Repository-Browser field from debian/upstream/metadata.
+  * Removed "-Wl,--as-needed" linker flag.
+  * Bumped Standards-Version (no changes necessary).
```
Comment (tsimonq2) [inline debian/changelog]: Technically it's `Bumped Standards-version to 4.6.1, [no changes needed / (no chances necessary).` Sorry for being anal. :hugs:

File: debian/changelog [hunk 22]
```diff
@@ -1,3 +1,14 @@
+libfm-qt (1.1.0-0ubuntu7~ppa2) jammy; urgency=medium
+
+  * Backport to Jammy.
+  * Updated Lubuntu Developers copyright.
```
Comment (tsimonq2) [inline debian/changelog]: Add more verbosity

File: debian/changelog [hunk 23]
```diff
@@ -1,3 +1,14 @@
...
+  * Updated symbols file
```
Comment (tsimonq2) [inline debian/changelog]: Add more verbosity

File: debian/changelog [hunk 24]
```diff
@@ -1,3 +1,14 @@
...
+  * Removed Repository-Browser field from debian/upstream/metadata.
```
Comment (tsimonq2) [inline debian/changelog]: Why did you remove it from there?

File: debian/changelog [hunk 25]
```diff
@@ -1,3 +1,14 @@
...
+  * Removed "-Wl,--as-needed" linker flag.
```
Comment (tsimonq2) [inline debian/changelog]: Why? Even saying `obsolete` goes a long way. Ditto for the above. :smile:

File: debian/changelog [hunk 26]
```diff
@@ -1,3 +1,14 @@
...
+  * Bumped Standards-Version (no changes necessary).
+
+ -- Aaron Rainbolt <arraybolt3@gmail.com>  Thu, 23 Jun 2022 21:47:05 -0500
```
Comment (tsimonq2) [inline debian/changelog]: Perhaps bump the date while you're at it?

File: debian/changelog [hunk 27]
```diff
@@ -205,7 +216,7 @@ libfm-qt (0.14.1-12ubuntu1) focal; urgency=medium
         trusting executables.
     - debian/patches: fix-dnd-mtp.patch
       + Fix drag-and-drop from mtp:// mounted folders.
-  
+
```
Comment (tsimonq2) [inline debian/changelog:220]: Thank you for doing wrap-and-sort. Fun fact: the only time I would never run that command is for anything that goes into the archive for a stable release. The processes are rigorous and that would be fair game for criticism. Just an FYI :smile:

File: debian/libfm-qt11.symbols [hunk 28]
```diff
@@ -368,6 +368,10 @@ libfm-qt.so.11 libfm-qt11 #MINVER#
  _ZN2Fm12FolderConfig10getIntegerEPKcPi@Base 0.16.0
  _ZN2Fm12FolderConfig10setBooleanEPKcb@Base 0.16.0
  _ZN2Fm12FolderConfig10setIntegerEPKci@Base 0.16.0
+ (arch=i386 armhf)_ZN2Fm12FolderConfig13getStringListEPKcPj@Base 1.1.0
+ (arch=!riscv64 !i386 !armhf)_ZN2Fm12FolderConfig13getStringListEPKcPm@Base 1.1.0
+ (arch=i386 armhf)_ZN2Fm12FolderConfig13setStringListEPKcPKS2_j@Base 1.1.0
```
Comment (tsimonq2) [inline debian/libfm-qt11.symbols:373]: Ditto

**Review by tsimonq2 — CHANGES_REQUESTED** [review]
First off, thank you for submitting this. I appreciate your willingness to go above and beyond to teach yourself symbols, that's one of the harder parts of packaging! Don't let the quantity of comments discourage you. I'm tired so it's much harsher than usual.

Once the comments are fixed I'll be happy to merge. BTW, you may consider rewinding to ppa1 if it isn't already in the PPA. If it is I need to know which commit to tag.

Thanks again.

File: debian/changelog (hunk 24 above)
Comment (ArrayBolt3) [reply debian/changelog]: Lintian griped about it, and in previous packages the exact identical data existed in the field above, so I've just been systematically removing that field for Lintian's sake. I get the feeling that *might* be bad? If so, we may have to redo a lot of my earlier backports.

File: debian/changelog (hunk 25 above)
Comment (ArrayBolt3) [reply debian/changelog]: Lintian gripe, removing it seems to work in other stuff and made Lintian calm down. I guess "obsolete" should be put in there.

File: debian/changelog (hunk 27 above)
Comment (ArrayBolt3) [reply debian/changelog:220]: Oh great. Alright, will learn to stop running that command before every. Single. Commit. When doing backports of archive stuff. (That's what I've been doing, didn't realize that would be a problem...)

File: debian/changelog (hunk 23 above)
Comment (ArrayBolt3) [reply debian/changelog]: Bear with me here, but... what else is there to be verbose about? I'm pretty sure I shouldn't mention all the steps I took to update the file, and given the horrific complexity of the file's guts, I don't want to go into detail about all the changes I made. Maybe you can suggest what extra verbosity to add?

**Comment by ArrayBolt3** [comment]
Not yet ready for merge, but that's a lot of stuff you pointed out fixed.

File: debian/changelog (hunk 26 above)
Comment (ArrayBolt3) [reply debian/changelog]: Done.

File: debian/changelog (hunk 25 above)
Comment (ArrayBolt3) [reply debian/changelog]: Done.

File: debian/changelog (hunk 24 above)
Comment (ArrayBolt3) [reply debian/changelog]: Added details.

File: debian/changelog (hunk 22 above)
Comment (ArrayBolt3) [reply debian/changelog]: Done.

File: debian/changelog (hunk 21 above)
Comment (ArrayBolt3) [reply debian/changelog]: Fixed.

**Comment by tsimonq2** [comment]
Thank you. As clarified async, running c++filt is just the first step. Those symbols will likely fail on RISC-V when the changes hit the archive, since on some lines you exclude three arches while you're including two arches on another line.

Impressive job with making the requested changes btw. You hit the nail on the head with the changelog. 😄

File: debian/copyright (hunk 20 above)
Comment (ArrayBolt3) [reply debian/copyright:20]: Confirmed, should I add it to the changelog?

**Comment by ArrayBolt3** [comment]
I believe this is done (the symbols are finally taken care of), however, as mentioned async, my RISC-V 64 build is giving me the following Lintian error: "E: libfm-qt11: binary-from-other-architecture usr/lib/riscv64-linux-gnu/libfm-qt.so.11.0.0" I do not know why this is happening, and I suspect Lintian is wrong, but double-check before merging.

**Comment by tsimonq2** [comment]
Merged to new branch.
